pgmpy/inference/base_continuous.py

```diff
@@ -0,0 +1,308 @@
+        # The gradient log of probability distribution at theta
+        self.log_pdf = None
+
+    def get_gradient_log_pdf(self):
+        return self.grad_log, self.log_pdf
+
+
+class GradientLogPDFGaussian(GradientLogPDF):
+    """
+    Class for finding gradient and gradient log of distribution
+    Inherits pgmpy.inference.base_continuous.GradientLogPDF
+    Here model must be an instance of JointGaussianDistribution
+    """
+
+    def __init__(self, theta, model):
+        GradientLogPDF.__init__(self, theta, model)
+        self.grad_log, self.log_pdf = self._get_gradient_log_pdf()
+
+    def _get_gradient_log_pdf(self):
+        """
+        Method that finds gradient and its log at theta
+        """
+        sub_vec = self.theta - self.model.mean_vec
+        grad = - np.dot(self.model.precision_matrix, sub_vec)
+        log_pdf = 0.5 * np.float(np.dot(sub_vec.T, grad))
+
+        return grad, log_pdf
+
+
+class DiscretizeTime(object):
+    """
+    Base class for Discretizing continuous time in sumilatin dynamics
+
+    Classes inheriting this base class can be passed as an argument for
+    discretizing time in inference algorithms
+
+    Parameters
+    ----------
+    theta : A 1d array type object or a row matrix of shape 1 X d
+            or d X 1.(Will be converted to d X 1)
+            Vector representing values of parameter theta( or X)
+
+    model : An instance of pgmpy.models.Continuous
+
+    grad_log_pdf : A subclass of pgmpy.inference.base_continuous.GradientLogPDF
+
+    momentum: A vector (row matrix or 1d array like object) of shape 1 X d
+              Vector representing the proposed value for momentum
+              (velocity)
+
+    epsilon: Float
+             step size for the time splitting
+
+    Examples
+    --------
+    >>> # Example shows how to use the container class
+    >>> from pgmpy.inference import DiscretizeTime
+    >>> from pgmpy.inference import GradLogPDFGaussian
+    >>> class ModifiedEuler(DiscretizeTime):
+    ...     def __init__(self, theta, model, grad_log_pdf, momentum, epsilon):
+    ...         DiscretizeTime.__init__(theta, model, grad_log_pdf, momentum, epsilon)
+    ...         self.theta_bar, self.momentum_bar, self.grad_log_theta,\
+    ...                 self.log_pdf = self._discretize_time()
+    ...     def _discretize_time(self):
+    ...         grad_log_theta, log_pdf_theta =\
+    ...                 self.grad_log_pdf(self.theta,
+    ...                                   self.model).get_gradient_log_pdf()
+    ...         momentum_bar = self.momentum + self.epsilon * grad_log_theta
+    ...         theta_bar = self.theta + self.epsilon * momentum_bar
+    ...        return theta_bar, momentum_bar, grad_log_theta, log_pdf_theta
+    """
+
+    def __init__(self, grad_log_pdf, model, theta, momentum, epsilon):
+        # TODO: Take model instead of precision matrix
+        if isinstance(theta, (np.matrix, np.ndarray, list, tuple, set, frozenset)):
```

Comment:
@ankurankan I think this is the cleanest way for doing this. Though I'll look for others also.